### ISSUE TITLE
Auto switch profiles based on AC/DC/PD

### ISSUE THREAD

**Issue by carverhaines** [CappyT/g14control#8]
Would be great to have a feature to set a default profile depending on your charging state, one for AC power adapter, one for DC battery, and one for USB-C Power delivery. Would need to be a user-defined variable in config.yaml.
Then have it auto-switch profiles depending on power state.

Example
```
app_name: "G14 Control"  # Name of the app, you can customize that as you wish!
notification_time: 3  # How long notifications will stay on the screen
check_power_every: 10  # Seconds in between checks to get the battery/ac status
default_ac_plan: "Performance"
default_dc_plan: "Silent (low-speed fan)"
default_pd_plan: "Silent (default fan speed)"
temp_dir: 'C:\temp\'
plans:
  - name: Silent (fanless)
    plan: silent
...
...
```

**Comment by CappyT** [comment]
Great idea!
It however requires the module to recognize the type of charger, which I'm not sure is possible from python.

**Comment by carverhaines** [comment]
Well, at least differentiate between charging and battery, and auto switch scripts based on user preference in config.yaml

**Comment by carverhaines** [comment]
Ok, I'm 90% of the way there, please see if you can figure out what I'm missing.
Here is my branch: https://github.com/carverhaines/g14control/tree/Issue8

Here is the traceback error, I believe I am calling apply_plan() wrong on lines 52 & 54
```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Users\user\AppData\Local\Programs\Python\Python38\lib\threading.py", line 932, in _bootstrap_inner
    self.run()
  File "C:\Users\user\AppData\Local\Programs\Python\Python38\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "C:\g14control\G14Control.pyw", line 54, in power_check
    apply_plan(default_dc_plan)
  File "C:\g14control\G14Control.pyw", line 202, in apply_plan
    current_plan = plan['name']
TypeError: string indices must be integers
```

**Comment by carverhaines** [comment]
And the current diff: https://github.com/CappyT/g14control/compare/master...carverhaines:Issue8

**Comment by carverhaines** [comment]
Test by running, and plugging & unplugging your AC adapter

**Comment by carverhaines** [comment]
@CappyT , please let me know how to pass an argument to apply_plan()
I've been trying to work it out all night!
Thanks!

**Comment by carverhaines** [comment]
Ok, finally figured it out. This one was quite complex.

All came down to figuring out what this mess means:
```
*list(map((lambda plan: pystray.MenuItem(plan['name'], (lambda: apply_plan(plan)))), config['plans'])), 
```
Basically it is passing the whole plan as a dictionary to apply_plan()

So first we have to iterate over the plans in config (which is a list of dictionaries) and search for the desired plan name, then send that to apply_plan() :
```
                for plan in config['plans']:
                    if plan['name'] == default_dc_plan:
                        break
                apply_plan(plan)
                print("finished applying DC plan: " + default_dc_plan)
```

Now automatic plan switching between AC/DC works.

However, you can't set a custom profile (right clicking menu), because in <10 sec the automatic plan switching overrides the chosen plan and switches back to the default. So now to work that one out...

**Comment by carverhaines** [comment]
Fixed in PR #26

**Comment by carverhaines** [comment]
Confirmed that PR also works with USB-C PD as well
